openfun/fun-apps
# Ficus rebase: Course Dashboard > Remove

Removes functionality from the FUN Course Dashboard which is provided by Ficus Insights.

**Screenshots**:

*Before this change:*

Course Dashboard Home:
![orig - dashboard home](https://user-images.githubusercontent.com/7556571/29155773-4029bc0e-7ddd-11e7-8ba1-3a67707459d9.png)

Course Dashboard Demo Course Home:
![orig - course home](https://user-images.githubusercontent.com/7556571/29155780-46e19012-7ddd-11e7-9a1d-cc32df1d7bde.png)

*After this change:*

Course Dashboard Home:
![removed - dashboard home](https://user-images.githubusercontent.com/7556571/29154873-ba656542-7dd6-11e7-9d0f-57e85ef9e86c.png)

Course Dashboard Demo Course Home:
![removed - course home](https://user-images.githubusercontent.com/7556571/29155696-db3d8cee-7ddc-11e7-93b9-ae10241913a2.png)

**Testing instructions**:

1. Check out this PR's branch under `/edx/app/edxapp/fun-apps`
1. Run the lms: 
   ```bash
    edxapp@precise64:~/edx-platform$ fun lms.dev run
    ```
1. Run the discussion forum: 
   ```bash
   forum@precise64:~/cs_comments_service$ ruby app.rb  -p 18080
    ```
1. Login to the LMS as the `admin@example.com` user, or a superuser. 
   I'm not sure what the default password for `admin@example.com` was, but I reset mine using the `fun lms.dev shell`.
1. Visit the Course Dashboard home page:
    http://localhost:8000/courses/fun/dashboard/
    Notice that the new home page is the Global Student Map, instead of the removed Global Enrollments page.
1. Visit the Course Dashboard page for the DemoX course:
   http://localhost:8000/courses/course-v1:edX+DemoX+Demo_Course/fun/dashboard/
   Notice that the new home page is the Course Forum Activity page, instead of the removed Course Enrollments page.
   Notice that the following links have been removed from the header: *Statistiques d'inscriptions*, *Carte des apprenants*, and *Répartition des réponses*.
1. Visit the Backoffice home page:
   http://localhost:8000/backoffice/
   Notice that the *Statistiques* link now directs you to the Global Student Map page, instead of the removed Global Enrollments page.
1. Visit the course detail page for the Demo course:
   http://localhost:8000/backoffice/course/course-v1:edX+DemoX+Demo_Course
   Notice that the *Tableau de bord* link now directs you to the Course Forum Activity page, instead of the removed Course Enrollments page.
1. Visit the removed pages, and note the 404 response:
    http://localhost:8000/courses/fun/dashboard/enrollments/
    http://localhost:8000/courses/course-v1:edX+DemoX+Demo_Course/fun/dashboard/enrollments/
    http://localhost:8000/courses/course-v1:edX+DemoX+Demo_Course/fun/dashboard/map/
    http://localhost:8000/courses/course-v1:edX+DemoX+Demo_Course/fun/dashboard/problem_stats/index/

**Author notes and concerns**:

The automated testing procedure is currently throwing a number of errors on the base `ficus` and `dev` branches:
```bash
edxapp@precise64:~/fun-apps$ git status
# On branch ficus
# Your branch is up-to-date with 'origin/ficus'.
#
nothing to commit, working directory clean
edxapp@precise64:~/fun-apps$ fun lms.test test
... 
Ran 318 tests in 59.825s
FAILED (SKIP=21, errors=10, failures=6)
```

This change removes the tests related to the removed functionality, and so the number of tests run decreases.  However, it does not fix these base issues, but we do ensure that no new errors appear.
```bash
edxapp@precise64:~/fun-apps$ git status
# On branch jill/ficus-ft-2-course-dashboard-remove
# Your branch is up-to-date with 'opencraft/jill/ficus-ft-2-course-dashboard-remove'.
#
nothing to commit, working directory clean
edxapp@precise64:~/fun-apps$ fun lms.test test
...
Ran 283 tests in 50.823s
FAILED (SKIP=21, errors=10, failures=6)
```

**Reviewers**
- [ ] FUN reviewer(s) TBD
